addons/website/static/src/client_actions/website_editor/website_editor.js

```diff
@@ -11,6 +11,39 @@ import {OptimizeSEODialog} from '@website/components/dialog/seo';
 
 const { Component, onWillStart, useEffect, useRef, useState } = owl;
 
+class BlockIframe extends Component {
+    setup() {
+        this.websiteService = useService('website');
+        this.state = useState({
+            blockIframe: false,
+            showLoader: false,
+        });
+        this.iframeLocks = 0;
+        this.websiteService.bus.addEventListener("BLOCK", this.block.bind(this));
+        this.websiteService.bus.addEventListener("UNBLOCK", this.unblock.bind(this));
+    }
+    block(event) {
+        if (event.detail.showLoader && !this.state.showLoader) {
+            this.state.showLoader = true;
+        }
+        if (this.iframeLocks === 0) {
+            this.state.blockIframe = true;
+        }
+        this.iframeLocks++;
+    }
+    unblock() {
+        this.iframeLocks--;
+        if (this.iframeLocks < 0) {
+            this.iframeLocks = 0;
+        }
+        if (this.iframeLocks === 0) {
+            this.state.blockIframe = false;
+            this.state.showLoader = false;
+        }
+    }
+}
+BlockIframe.template = 'website.BlockIframe';
+
 export class WebsiteEditorClientAction extends Component {
```

Comment:
@younn-o I added the iframeblock component to the same file as the website_editor client action. To me it made sense to not add extra files but also because the iframe is defined inside that component. Is that ok with you?

Reply:
It makes sense, the component will probably not be used anywhere else :)